### PR TITLE
Alertmanager configuration validate-only mode #3437

### LINKED ISSUE #3437: Feature request: A command line mode be made to validate alertmanager configuration
#### Is your feature request related to a problem? Please describe.

Our mimir configuration is done through a CI system, and is deployed out to multiple instances in multiple environments. We have validations for many items before a new change can go into that. 

Performing `alertmanager load` clearly validates before the configuration is uploaded to a server, however it can break if an incorrect config has been merged into the main branch. We do not have a way to validate this config before this is merged.

#### Describe the solution you'd like

It would useful for validation of changes into our deployable main, to have something like `mimirtool alertmanager verify <config>` that would perform the validation, but without requiring to upload to a server afterwards.

#### Describe alternatives you've considered

Building a bash script around `alertmanager load` that will filter out an error due to `ERRO[0000] error during request to Grafana Mimir API` and give it dummy URI's and credentials.

#### Additional context

N/A

### PR DESCRIPTION
**Pull request by dannystaple**
#### What this PR does

This introduces an additional flag for `mimirtool alertmanager load` to validate configuration only, without proceeding to upload it to server. This can be used to validate configuration, for example in PR CI pipelines before attempting to merge and deploy.

#### Which issue(s) this PR fixes or relates to

Fixes #3437

#### Checklist

- [ ] Tests updated - no tests exist for `mimirtool alertmanager load`
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/sources/operators-guide/architecture/components/alertmanager.md
Comment (osg-grafana) [inline docs/sources/operators-guide/architecture/components/alertmanager.md:27]:
```suggestion
You can validate a configuration file using the `mimirtool` command:
```

Comment (osg-grafana) [reply docs/sources/operators-guide/architecture/components/alertmanager.md:27]: Rationale: _may_ connotes permission; _can_ connotes possibility.

**Review by osg-grafana — APPROVED** [review]
A nit

**Comment by Logiraptor** [comment]
I liked the suggestion you had in the issue about `mimirtool alertmanager verify` (or maybe `mimirtool alertmanager validate`. Is there a reason that isn't an option? I think it would be preferable to have an explicit command to avoid the need to specify address and tenant when it's not used. What do you think?

**Comment by dannystaple** [comment]
> I liked the suggestion you had in the issue about `mimirtool alertmanager verify` (or maybe `mimirtool alertmanager validate`. Is there a reason that isn't an option? I think it would be preferable to have an explicit command to avoid the need to specify address and tenant when it's not used. What do you think?

Let me take a look at that. Certainly sounds reasonable as the id/address is a little ugly.

**Comment by dannystaple** [comment]
Ok - committed `mimirtool alertmanager verify`.

File: CHANGELOG.md
Comment (pracucci) [inline CHANGELOG.md:100]: [nit] I think you can proudly call it a new feature ;) We're also used to reference the PR number, so #3440 in this case.

```suggestion
* [FEATURE] Added `mimirtool alertmanager verify` command to validate configuration without uploading. #3440
```

File: docs/sources/operators-guide/tools/mimirtool.md
Comment (pracucci) [inline docs/sources/operators-guide/tools/mimirtool.md:95]: The `##### Example`  below is related to "Load Alertmanager configuration". I think you should:
1. Move `### Validate Alertmanager configuration` after the example below (right before `#### Delete Alertmanager configuration`)
2. Change `### Validate Alertmanager configuration` to `#### Validate Alertmanager configuration` (4 hashes) to keep formatting consistent

File: pkg/mimirtool/commands/alerts.go
Comment (pracucci) [inline pkg/mimirtool/commands/alerts.go:74]: [nit] Since there's a bit of repetition here between `getAlertsCmd`, `deleteCmd` and `loadalertCmd`, what if we register these flags in a for loop, something like (pseudocode):

```
for _, cmd := range []*kingpin.CmdClause{getAlertsCmd, deleteCmd, loadalertCmd } {
   cmd.Flag("address", ...)
   cmd.Flag("id", ...)
}
```

File: pkg/mimirtool/commands/alerts.go
Comment (pracucci) [inline pkg/mimirtool/commands/alerts.go:88]:
```suggestion
	vertifyalertCmd.Arg("config", "Alertmanager configuration to verify").Required().StringVar(&a.AlertmanagerConfigFile)
	vertifyalertCmd.Arg("template-files", "The template files to verify").ExistingFilesVar(&a.TemplateFiles)	
```

File: pkg/mimirtool/commands/alerts.go
Comment (pracucci) [inline pkg/mimirtool/commands/alerts.go:116]: [nit] WDYT about calling it "read"? We're basically reading it from file.

```suggestion
func (a *AlertmanagerCommand) readAlertmanagerConfigFromFile(k *kingpin.ParseContext) (error, string, map[string]string) {
```

**Review by pracucci — COMMENTED** [review]
Good job! I left few minor comments before merging. Thanks!

File: CHANGELOG.md
Comment (dannystaple) [reply CHANGELOG.md]: Done. :)

File: pkg/mimirtool/commands/alerts.go
Comment (dannystaple) [reply pkg/mimirtool/commands/alerts.go]: I like that - more DRY. Done.

File: pkg/mimirtool/commands/alerts.go
Comment (dannystaple) [reply pkg/mimirtool/commands/alerts.go]: Agreed. And done.

**Review by jhesketh — APPROVED** [review]
lgtm

File: pkg/mimirtool/commands/alerts.go
Comment (pracucci) [inline pkg/mimirtool/commands/alerts.go:85]: [nit] Doh, there's a small typo here:

```suggestion
	verifyalertCmd := alertCmd.Command("verify", "Verify Alertmanager tenant configuration and template files.").Action(a.verifyAlertmanagerConfig)
```

**Review by pracucci — APPROVED** [review]
LGTM (modulo a nit). Thanks!

File: pkg/mimirtool/commands/alerts.go
Comment (dannystaple) [reply pkg/mimirtool/commands/alerts.go]: Ok - fixed.

**Pull request merged**
